client/state/form/reducer.js

```diff
@@ -34,9 +27,13 @@ export default function form( state = {}, action ) {
 		} );
 	}
 
-	// Allow client-side form validation to take over error state when inputs change
-	if ( settingsActions.hasOwnProperty( action.type ) && newState.hasOwnProperty( 'errors' ) ) {
-		delete newState.errors;
+	if ( settingsActions.hasOwnProperty( action.type ) ) {
+		newState.pristine = false;
```

Comment:
Ah - I see. Yeah, that action name is poor.  Wanna change it to SET_FORM_PROPERTY or something instead?